includes/file.form.inc

```diff
@@ -21,7 +21,8 @@ function islandora_marcxml_file_form(array $form, array &$form_state) {
     'file' => array(
       '#type' => 'managed_file',
       '#title' => t('MARCXML File'),
-      '#description' => t('A file containing a MARCXML record, to be transformed to MODS. Click "Next" to skip this step and create a record from scratch.'),
+      '#description' => t('A file containing a MARCXML record, to be transformed to MODS. Click "Next" to skip this step and create a record from scratch.
+        To disable this ingest step, <a href="/admin/islandora/tools/islandora_marcxml">configure the Islandora MARCXML module.</a>'),
```

Comment:
Linking to an admin page for users that may not have the appropriate permissions here. Also this is not a multisite/language safe way to create a link.

Reply:
@willtp87 
> Linking to an admin page for users that may not have the appropriate permissions here. 
Is it a problem to link to a page that a user might not have permission to access? 

> Also this is not a multisite/language safe way to create a link.
Multisite-safe: How so? It works in my multisite.
Language-safe: OK, trying to make this work, but am failing:

` ``
 '#description' => t('A file containing a MARCXML record, to be transformed to MODS. Click "Next" to skip this step and create a record from scratch.
        To disable this ingest step, <a href="!url">configure the Islandora MARCXML module.</a>', '!url' => url('/admin/islandora/tools/islandora_marcxml'),
```

What am I doing wrong there?

Reply:
@bondjimbond
It is a problem to link to known preventable 401s. It is not good UX. Imagine an unauthenticated user performing ingest and clicking this link to get a 401. They may simply never submit to the site again. This can be prevented with an access check.
Multisites can use a similar prefix as languages and have the same issues with hard coded URLs.
I believe the second parameter to your `t()` call should be more like `array('@url' => url('admin/islandora/tools/islandora_marcxml')`. Notice the array syntax, lack of a leading slash, and the use of the `@`.

Reply:
> I believe the second parameter to your t() call should be more like array('@url' => url('admin/islandora/tools/islandora_marcxml'). Notice the array syntax, lack of a leading slash, and the use of the @.

I tried that too, still getting a syntax error.

> syntax error, unexpected '=>' (T_DOUBLE_ARROW) in includes/file.form.inc on line 27

Reply:
@bondjimbond  if you push your code somewhere I can probably spot your syntax error.

Reply:
https://github.com/bondjimbond/islandora_marcxml/blob/bad_syntax/includes/file.form.inc#L25

Reply:
That line is missing the closing parenthesis on the array and the call to `url()`.

Reply:
Thanks! Such a simple thing, so easy to miss... and the error message of course misdiagnoses it.

Reply:
As to your other comment:

> It is a problem to link to known preventable 401s. It is not good UX. Imagine an unauthenticated user performing ingest and clicking this link to get a 401. They may simply never submit to the site again. This can be prevented with an access check.

How would you approach this?

Reply:
I would throw the help text into its own element and use [#access](https://api.drupal.org/api/drupal/developer%21topics%21forms_api_reference.html/7.x#access) with a [check](https://api.drupal.org/api/drupal/modules!user!user.module/function/user_access/7.x) on the appropriate [admin perm](https://github.com/Islandora/islandora_marcxml/blob/7.x/islandora_marcxml.module#L49) but there are of course other ways.

Reply:
Alternatively, what if we instruct the user to configure the module but do not include a link?

Reply:
Instructing a user to configure something that they don't have access to is just a slightly less bad instance of the same issue as the link. That's why you will often see 'Please contact your administrator.' messages.

Reply:
So would it be acceptable to say something like "To disable this ingest step, an administrator must configure the Islandora MARCXML module"?

Reply:
That would be better but I'm not sure it is appropriate to suggest to every user they should be thinking about disabling feature. It is unnecessary cognitive load regarding something they can not take direct action about.  If any message is present I think it should be shown only to those that it applies to.

Reply:
@bondjimbond @willtp87 do we need a message here? I feel this is just a documentation issue, i would rather not expose this to someone just ingesting, that could not at all have permissions to enable to toggle, or even an admin that has made its mind already of having this step and really does not need a constant reminder. I feel this is pushing the need for information a bit too much

Reply:
@DiegoPino  I agree.

Reply:
I like the additional help text. That ingest step may not be commonly seen in production, but it's part of the 'demos' of Islandora that people access from our VM or vagrant, and it can be confusing/irritating. Letting people know where it comes from and how to get rid of it would be nice for those cases.

Reply:
@manez  That sort of use case usually handled through a custom module or theme.

Reply:
So it sounds like we... don't want a message at all, is that right?

I prefer a message, because I've got a lot of admins under me who are nominally running their own sites, but form whom I provide the majority of support. And those ingesting the objects, while they might not be able to configure the module themselves, would be very happy to know that the setting CAN be turned off and they just need to ask for it.

Reply:
@willtp87 @DiegoPino @manez Any further thoughts on the message/no message question, given my note above?

Reply:
@bondjimbond custom help messages are not uncommon. I don't think that this message is appropriate for most users. I do not consider this a blocker.

Reply:
I'm still in favor of it - I don't see the harm in having more documentation. But I don't feel strongly enough about it to block either.

Reply:
As someone who administers a bunch of sites with administrators and users of different skill levels, I think a message is good.

So if I include this message, with no filtering for privileges:

> To disable this ingest step, an administrator must configure the Islandora MARCXML module.

Will you approve?

Reply:
Now updated to an update hook message.